### ISSUE TITLE
removeAttribute creates a dirty model

### ISSUE THREAD

**Issue by thomasedwards** [baopham/laravel-dynamodb#152]
**Describe the bug**

When using `removeAttribute()`, followed by `save()`, it will save the model as it was before.

```php
    $model = new Model;
    $model->oldThing = true;
    $model->save();

    // Later...

    $model = new Model;
    $model->newThing = true;
    $organisation->removeAttribute('oldThing');
    $model->save();

    // $model has oldThing and newThing

    // Later...

    $model = new Model;
    $organisation->removeAttribute('oldThing');
    $model->save();

    // $model has oldThing and newThing
```

In [Eloquent](https://github.com/laravel/framework/blob/5.6/src/Illuminate/Database/Eloquent/Model.php), the model checks to see if it’s dirty before saving:

```php
    // If the model already exists in the database we can just update our record
    // that is already in this database using the current IDs in this "where"
    // clause to only update this model. Otherwise, we'll just insert them.
    if ($this->exists) {
        $saved = $this->isDirty() ?
                    $this->performUpdate($query) : true;
    }
```

Which would solve the case where `removeAttributes()` is run and then a `save()` is performed anyway, such as on an `update`-type of request, but it wouldn’t solve the adding/removing at the same time problem.

**Version info**

* Laravel: 5.6
* Library: latest

**Comment by baopham** [comment]
Awesome catch! A fix will come in a few days.

**Comment by thomasedwards** [comment]
Thanks for replying to the issue @baopham! I’ve worked around it myself by designing the removeAttributes use to only be used as a one-off action which isn’t too bad, but will be very grateful if you solve it!

Yeah, sounds really tricky. I hadn’t thought of using `refresh()`, but I can see what you mean.

**Comment by Tamrael** [comment]
> So essentially after calling `removeAttributes`, we should expect that the model would have the fresh value (i.e. the attributes should be removed from the instance). The easiest way is to call `refresh()` but I don't want to do that, which means this is going to take a bit more time than I expected.

referring to [this](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.REMOVE) part of the dynamodb documentation you can just add `--return-values ALL_NEW` to the query and you will get the item as it appears in dynamodb after the removeAttribute operation. Then one would only need to reapply the previously "dirty" parts of the model to not overwrite changes. This way there won't be a need for additional queries like refresh() would issue

**Comment by baopham** [comment]
This is perfect @Tamrael! Much easier then what I had in mind

**Comment by baopham** [comment]
Fixed in v4.11.1

**Comment by thomasedwards** [comment]
Just removed 7 lines of code already from one project 👌

Thank you very much!
